### DISCUSSION TITLE
massDOT Example API Requirements Document

### DISCUSSION THREAD

**Discussion by wesleyalforddot** [usdot-jpo-ode/jpo-wzdx#121] [Show and tell]
The attached requirements document, provided by MassDOT, defines the API that must be implemented by any Smart Work Zone (SWZ) vendor so that it may be consumed by an agency’s SWZ manager software. This document provides clearly articulated requirements and best practices, and may be helpful for states looking to initiate their own WZDx data feed. It is intended for data management personnel who are responsible for writing their jurisdiction’s APIs or involved in procuring contracts for roadway construction. Topics covered within the document include:

- **Security** – the username/password authentication system used between the client (an agency’s SWZ management software) and server (the SWZ vendor’s field application).

- **Work zone definition** – descriptions of terminology to be included in the API, aligned with the WZDx specification and the TMDD standard.
- **API requests** – Defines a REST API with methods for interacting with endpoint summary, vendor information, road events, field device configuration, road event dynamic metrics, and field device data, among others. Also includes example calls with success and failure responses.  

[massDOT - Vendor API v5.0.pdf](https://github.com/usdot-jpo-ode/jpo-wzdx/files/5029316/massDOT.-.Vendor.API.v5.0.pdf)

**Comment by GISProPasco** [comment]
Incredibly useful, thank you all so very much. Working towards being able to contribute. Thanks again.

**Comment by bdelsey64** [comment]
All

Please note that the referenced document is not the latest/proper version. The link below is correct.

Regards
Brian Delsey

[MassDOT - Vendor API v6.0.pdf](https://github.com/usdot-jpo-ode/jpo-wzdx/files/5029877/MassDOT.-.Vendor.API.v6.0.pdf)

**Reply by wesleyalforddot** [reply]
Thanks for the latest version! Feel free to add any additional information that you think would be helpful to guide individuals through the document.
